examples/word-count/test/word_count_e2e_test.clj

```diff
@@ -0,0 +1,111 @@
+
+(defn word-watcher
+  "Builds a test-command that blocks until the supplied word appears"
+  [word]
+  [:watch (fn [journal]
+            (some #(= word (:key %))
+                  (get-in journal [:topics :output]))) 2000])
+
+(defn wc
+  "A simple helper to extract the latest value from the word-count ktable
+   as observed by the test-consumer.
+
+   The journal collects all records as a vector of maps representing
+   ConsumerRecords for each topic. Since we're inspecting a mutating
+   table, we want to get the `last` matching record for `word`."
+  [journal word]
+  (->> (get-in journal [:topics :output])
+       (filter (fn [r]
+                 (= (:key r) word)))
+       last
+       :value))
+
+(defn get-env [k]
+  (get (System/getenv) k))
+
+(def test-config
+  {:broker-config broker-config
+   :topic-config wc/topic-metadata
+   :kstream-config (assoc wc/app-config "cache.max.bytes.buffering" "0")
+   :enable? (get-env "BOOTSTRAP_SERVERS")})
+
+(defn props-for [x]
+  (doto (Properties.)
+    (.putAll (reduce-kv (fn [m k v]
+                          (assoc m (str k) (str v)))
+                        {}
+                        x))))
+
+(defn mock-transport-config
+  []
+  {:driver (let [builder (k/streams-builder)
+                 app (wc/topology-builder wc/topic-metadata)
+                 topology (.build (proto/streams-builder* (app builder)))]
+             (TopologyTestDriver.
+              topology
+              (props-for (:kstream-config test-config))))})
+
+(defn test-transport
+  [topics]
+  (cond
+    (get-env "REST_PROXY_URL")
+    (let [rest-proxy-url (get-env "REST_PROXY_URL")]
+      (jd.test/rest-proxy-transport {:bootstrap-uri rest-proxy-url
+                                     :group-id (get (test-consumer-config {}) "group.id")}
+                                    topics))
+
+    (get-env "BOOTSTRAP_SERVERS")
+    (let [broker (get-env "BOOTSTRAP_SERVERS")]
+      (jd.test/kafka-transport (test-consumer-config {"bootstrap.servers" broker})
+                               topics))
+
+    :else
+    (jd.test/mock-transport (mock-transport-config) wc/topic-metadata)))
+
+(deftest test-word-count-demo
+  (fix/with-fixtures [(fix/integration-fixture wc/topology-builder test-config)]
+    (fix/with-test-machine (test-transport wc/topic-metadata)
+      (fn [machine]
+        (let [lines ["As Gregor Samsa awoke one morning from uneasy dreams"
+                     "he found himself transformed in his bed into an enormous insect"
+                     "What a fate: to be condemned to work for a firm where the"
+                     "slightest negligence at once gave rise to the gravest suspicion"
+                     "How about if I sleep a little bit longer and forget all this nonsense"
+                     "I cannot make you understand"]
+              commands (->> (concat
+                             (map input-writer lines)
+                             [(word-watcher "understand")]))
```

Comment:
Can the commands be written in pure EDN? Is it worth doing that?

Reply:
It was like that in the first iteration. This seemed clearer IMO.